### PR TITLE
Test dataset loss added.

### PR DESCRIPTION
**Pull request by daviddiazvico**
Optional test dataset added to the fit method of both Sequential and Graph models. If a validation dataset is used, the test dataset can also be supplied to monitor the loss function.

### CODE REVIEW

**Comment by fchollet** [comment]
In Keras, the "validation data" is set of samples that is excluded from training, and on which we monitor performance during training. Having two such sets serves no purpose over having one.

**Comment by daviddiazvico** [comment]
I think the point is that the validation set can be used to do early stopping (via the EarlyStopping Callback) to prevent overfitting, while the test set can never change training (discussed here http://stats.stackexchange.com/questions/19048/what-is-the-difference-between-test-set-and-validation-set). Sometimes I find it useful to monitor this loss (and plot it along validation and train losses) during training, although it doesn't affect the results.

In any case, I will not insist with this.
